integration/tests/basic/test_contract_reverting.py

```diff
@@ -134,3 +137,31 @@ def test_custom_error_revert(self, revert_contract):
     def test_assert_revert(self, revert_contract):
         with pytest.raises(web3.exceptions.ContractPanicError, match="Panic error 0x01: Assert evaluates to false"):
             revert_contract.functions.doAssert().call()
+
+    @pytest.mark.slow
+    @pytest.mark.timeout(40 * Time.MINUTE)
+    def test_reverted_tx_receipt_after_30_minutes(self, revert_contract, json_rpc_client):
+        sender_account = self.accounts[0]
+        # instruction_tx = revert_contract.functions.doTrivialRevert().build_transaction(
+        instruction_tx = revert_contract.functions.customErrorRevert(1, 2).build_transaction(
+            {
+                "from": sender_account.address,
+                "nonce": self.web3_client.eth.get_transaction_count(sender_account.address),
+                "gasPrice": self.web3_client.gas_price(),
+                "gas": 100000000,
```

Comment:
Why do you set gas manually?

Reply:
This is how build transaction works — we need to prevent gas estimation to build and send transaction with broken function, that will be reverted.